homeassistant/components/mqtt/client.py

```diff
@@ -905,17 +908,32 @@ async def _async_perform_subscriptions(self) -> None:
         self._pending_subscriptions = {}
 
         subscription_list = list(subscriptions.items())
-        result, mid = self._mqttc.subscribe(subscription_list)
 
-        if _LOGGER.isEnabledFor(logging.DEBUG):
-            for topic, qos in subscriptions.items():
-                _LOGGER.debug("Subscribing to %s, mid: %s, qos: %s", topic, mid, qos)
-        self._last_subscribe = time.monotonic()
+        def split_subscribe_chunks(
+            subscription_list: list[tuple[str, int]],
+        ) -> Generator[list[tuple[str, int]], None, None]:
+            """Split the subscription list in chunks to avoid buffer issues."""
+            for chunk_index in range(
+                0, len(subscription_list), MAX_SUBSCRIBES_PER_CALL
+            ):
+                yield subscription_list[
+                    chunk_index : chunk_index + MAX_SUBSCRIBES_PER_CALL
+                ]
 
-        if result == 0:
-            await self._async_wait_for_mid(mid)
-        else:
-            _raise_on_error(result)
+        for chunk in split_subscribe_chunks(subscription_list):
```

Comment:
We could probably move https://github.com/home-assistant/core/blob/e7d23d8b4963f87fd59d40c6ca3d40a476c0a0c7/homeassistant/components/recorder/util.py#L878 into a core helper/util and use it here

Reply:
done